### PR TITLE
Update PHPDoc

### PR DESCRIPTION
**Pull request by FrontEndCoffee**
I went ahead and updated some of the php-doc-blocks. I also made sure that all the possible typehints are applied (keeping the php 5.5 version constraint in mind).

### CODE REVIEW

File: src/RetryMiddleware.php [hunk 1]
```diff
@@ -90,6 +96,12 @@ private function onFulfilled(RequestInterface $req, array $options)
         };
     }
 
+    /**
+     * Execute rejected closure
+     * @param  RequestInterface $req
+     * @param  array            $options
+     * @return mixed
```
Comment (BackEndTea) [inline src/RetryMiddleware.php]: Doesn't this return a closure?

File: src/functions.php [hunk 2]
```diff
@@ -60,7 +60,7 @@ function describe_type($input)
  *                     format: "Name: Value"
  * @return array
  */
-function headers_from_lines($lines)
```
Comment (BackEndTea) [inline src/functions.php]: Currently this can take any iterable like a `Generator` etc. Adding the array type will be a BC break

Comment (FrontEndCoffee) [reply src/functions.php]: I fixed this

File: src/RetryMiddleware.php (hunk 1 above)
Comment (FrontEndCoffee) [reply src/RetryMiddleware.php]: Changed it to callable

**Comment by gmponos** [comment]
Some of the functions have a docblock on the interface. So they inherit the one from the interface by default. What is the value of adding one on the functions as well?

**Comment by FrontEndCoffee** [comment]
@gmponos in my opinion it's much nicer to work with code that has docblocks on every method, even though it might be a duplicate of its interface.

It has the added benefit that most of the IDE's can interpret those docblocks.

File: src/Client.php [hunk 3]
```diff
@@ -89,6 +95,15 @@ public function __call($method, $args)
             : $this->request($method, $uri, $opts);
     }
 
+    /**
+     * Asynchronously send an HTTP request.
+     *
+     * @param RequestInterface $request Request to send
```
Comment (Nyholm) [inline src/Client.php]: "Request to send" is not needed.

**Comment by Nyholm** [comment]
This PR was merged in #2384

**Pull request closed without merging**